### PR TITLE
Add GHA build

### PR DESCRIPTION
**Pull request by mosabua**
I am not an expert of this .net stuff .. just following what I find at https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-net and cobbling something together

Or maybe https://github.com/microsoft/setup-msbuild

Also have to figure out https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry and https://github.com/NuGet/setup-nuget

Still very much a work in progress

Do we have to merge this so it even runs in the first place? At this stage I am flying blind .. going to get back to this after some feedback and time to look into it more.

Probably also have to figure out how install this stuff locally...

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up dotnet ${{ matrix.dotnet-version }}
```
Comment (nineinchnick) [inline .github/workflows/ci.yml:14]: There's no matrix yet

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Set up msbuild
+        uses: microsoft/setup-msbuild@v2
+      - name: Set up nuget
+        uses: nuget/setup-nuget@v2
+        with:
+          nuget-version: '5.x'
+      - name: Run mnsbuild
+        run: msbuild trino-csharp/TrinoDriver.sln   
+      - name: Run nuget for client
+        run: nuget pack Trino.Client\Trino.Client.nuspec
```
Comment (nineinchnick) [inline .github/workflows/ci.yml:25]: This should go into a release workflow.

**Review by nineinchnick — APPROVED** [review]
Let's merge this and iterate

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mosabua) [reply .github/workflows/ci.yml]: I thought we just set it here for starters .. but fine.. removed for now.. one step at a time.

**Comment by mosabua** [comment]
I am merging this to see what happens ;-)

**Pull request merged**